### PR TITLE
OP-372 Operation results list return code instead of translations

### PR DESCRIPTION
**Pull request by gildasdongmo**

### CODE REVIEW

File: src/main/java/org/isf/operation/manager/OperationBrowserManager.java [hunk 1]
```diff
@@ -144,11 +144,11 @@ public boolean descriptionControl(String description, String typeCode) throws OH
...
 	 */
-	public ArrayList<String> getResultsList() { //TODO (OP-372): deal with codes instead of translations!!!
+	public ArrayList<String> getResultsList() { 
 		ArrayList<String> resultsList = new ArrayList<String>();
-		resultsList.add(MessageBundle.getMessage("angal.operation.result.sucess"));
-		resultsList.add(MessageBundle.getMessage("angal.operation.result.failure"));
-		resultsList.add(MessageBundle.getMessage("angal.operation.result.undefined"));
+		resultsList.add("angal.operation.result.sucess");
+		resultsList.add("angal.operation.result.failure");
+		resultsList.add("angal.operation.result.undefined");
```
Comment (mwithi) [inline src/main/java/org/isf/operation/manager/OperationBrowserManager.java:151]: fix: sucess -> success

Anyway, with this method, in the DB will finish the whole string "angal.operation.result.sucess":

image

What about using a mapping mechanism as done in (e.g.) [ExaminationBrowserManager.java:92](https://github.com/informatici/openhospital-core/blob/develop/src/main/java/org/isf/examination/manager/ExaminationBrowserManager.java#L92) ?

**Review by mwithi — CHANGES_REQUESTED** [review]
to fix bundle name and to revise the mapping to the DB.

in any case the migration script [step_57_multiple_operations_in_admission.sql](https://github.com/informatici/openhospital-core/blob/develop/mysql/db/step_57_multiple_operations_in_admission.sql) needs to be changed accordingly

**Comment by mwithi** [comment]
@gildasdongmo +1

File: src/main/java/org/isf/operation/manager/OperationBrowserManager.java (hunk 1 above)
Comment (gildasdongmo) [reply src/main/java/org/isf/operation/manager/OperationBrowserManager.java:151]: > What about using a mapping mechanism as done in (e.g.) [ExaminationBrowserManager.java:92](https://github.com/informatici/openhospital-core/blob/develop/src/main/java/org/isf/examination/manager/ExaminationBrowserManager.java#L92) ?

It was just a choice, I can do it like that if necessary.

Comment (mwithi) [reply src/main/java/org/isf/operation/manager/OperationBrowserManager.java:151]: I think it's better as pattern; also step_57_multiple_operations_in_admission.sql should be changed after.
Thanks!

Comment (gildasdongmo) [reply src/main/java/org/isf/operation/manager/OperationBrowserManager.java]: Hi @mwithi, I made a new commit that uses the mapping mechanism as requested. and for what needs to be saved in bd, I used the keys already present in step_57_multiple_operations_in_admission.sql file and I think it won't be necessary to update it anymore.

File: src/main/java/org/isf/operation/manager/OperationBrowserManager.java [hunk 2]
```diff
@@ -144,11 +150,39 @@ public boolean descriptionControl(String description, String typeCode) throws OH
...
+	public LinkedHashMap<String, String> getResultsList() { 
+		if (resultsListHashMap == null) buildResultHashMap();
+		return resultsListHashMap;
+	}
+	
+	private void buildResultHashMap() {
+		resultsListHashMap = new LinkedHashMap<>();
+		resultsListHashMap.put("Success", MessageBundle.getMessage("angal.operation.result.success"));
+		resultsListHashMap.put("Failure", MessageBundle.getMessage("angal.operation.result.failure"));
+		resultsListHashMap.put("Unknown", MessageBundle.getMessage("angal.operation.result.undefined"));
```
Comment (mwithi) [inline src/main/java/org/isf/operation/manager/OperationBrowserManager.java:162]: as "codes" in the DB I would put them lowercase

**Review by mwithi — CHANGES_REQUESTED** [review]
It looks fine, but:
- let's use lowercase "keys" in the db
- please resolve the conflicts
- please apply imports standards (see README.md)

File: src/main/java/org/isf/operation/manager/OperationBrowserManager.java (hunk 2 above)
Comment (gildasdongmo) [reply src/main/java/org/isf/operation/manager/OperationBrowserManager.java]: > * let's use lowercase "keys" in the db
> * please resolve the conflicts
> * please apply imports standards (see README.md)

**All done.**

> in any case the migration script [step_57_multiple_operations_in_admission.sql](https://github.com/informatici/openhospital-core/blob/develop/mysql/db/step_57_multiple_operations_in_admission.sql) needs to be changed accordingly

script has been updated using lowercase **keys**

**Review by mwithi — APPROVED** [review]

**Comment by mwithi** [comment]
LGTM.
```
Results :

Tests run: 1105, Failures: 0, Errors: 0, Skipped: 1
```

**Pull request merged**